# Using a Machine Learning Model to Predict Plant Inflorescences based upon its Soil Microbiome
By Luke Denoncourt

## Introduction
The UN estimated that the global population could reach 9.7 billion by 2050 (United Nations). As a result, the amount of food required to feed humanity is thought to double by 2050 (Ray et al., 2012); whereas other studies forecast a 50% to 75% required increase in certain areas of crop production (Prosekov & Ivanova, 2018). From 1985 to 2005, the total amount of crops produced increased by 28%. However, crop yields have begun to stagnate, and even decline, in some parts of the world (Ray et al., 2012). Additionally, climate change threatens current crop production and is expected to cause crop loss in the US, with corn, soybean, and cotton production expected to decrease by 30% to 36% (Schlenker & Roberts, 2009) Cai, Wang, and Laurent (2009) predict the rainfed corn yield in central Illinois to decrease by 23% to 34% by 2055. With the increase in demand, and predicted decline in yields, humanity must find a way to increase crop production. One way to do so is by leveraging the interactions between the soil microbiome and the host plant.

Plant microbiomes have been shown to increase plant health, by contributing to disease suppression (Peralta et al., 2018; Wei et al., 2019), growth and development (Panke-Buisse et al., 2016), and nutrient uptake (Taffner et al., 2020; Trivedi et al., 2020). Plant growth promoting rhizobacteria have been shown to increase many facets of plant health, with the increased uptake of nutrients thought to arise from stimulation of root formation leading to bigger root systems with more root hairs (Adesemoye & Kloepper, 2009). Arbuscular mycorrhiza fungi (AMF) have been shown to increase phosphorus uptake in plants. By engineering soils comprising these beneficial microbes, researchers can create crops with higher yields.

Using potted plants with soils inoculated with _Pseudomonas alcaligenes, Pseudomonas denitrificans, Bacillus polymyxa, and Mycobacterium phlei,_ both the shoot and root growth of pea and cotton plants was significantly increased (Egamberdieva & Höflich, 2004). These artificial microbiomes, curated by the researchers, were shown to increase plant health. However, _in vivo,_ these communities are dynamic and affected by many factors, including the host plant. With the vast array of microbial taxa affecting the plant directly, indirectly, and through combinations of other interactions, it is important to disentangle and identify the microorganisms contributing to these beneficial interactions. This identification can be made with machine learning models.

Researchers have utilized machine learning models with microbiomes as the input features to predict important host health traits (Chang et al., 2017). The researchers used a Random Forest model to predict whether an area in a crop field was a high or low productive plot by examining the soil microbiome. After investigating the soil characteristics and using a logistic regression that showed no association with crop productivity and 26 tested soil characteristics, the researchers ran a PCA that showed a difference in the soil microbiome community composition between high and low productive fields. The Random Forest model, evaluating the soil microbiome at the taxonomic level of Order, had an accuracy of 79% when predicting field productivity. The model also identified multiple nitrogen utility-related taxa as important to making the prediction, such as the phyla Actinobacteria, Proteobacteria, and Cyanobacteria.

# ***TALK ABOUT RANDOM FOREST MODELS HERE IN INTRO***
These studies have shown that machine learning can be used to identify potential interactions between the soil microbiome and the host. These interactions can then be further explored to determine their effects on the host plant and what artificially created microbiome will create healthier plants. **In this study, I will use the soil microbiome as the features in a machine learning model to predict whether an Asclepias plant has zero or 1 or more inflorescences**


## Methods
### *Sample Collection*
Samples of Asclepias (Common Milkweed) were gathered in the summers of 2020 and 2021 in locations throughout Virginia (Figure 1).

![**Figure 1: Map of sampling locations across Virginia.** In total, 14 sites were used in this analysis. Sites were sampled over the summer months in 2020 and 2021.](https://user-images.githubusercontent.com/67921793/167645480-bd08d0fc-602c-46c6-97b0-cdede650a098.png)
**Figure 1: Map of sampling locations across Virginia.** In total, 14 sites were used in this analysis. Sites were sampled over the summer months in 2020 and 2021.

At each site, the _Asclepias_ plants were measured for phenotypic traits, such as height and number of leaves, and soil cores were taken near the plant, but far enough to minimize root mass being collected in the soil core. The leaf and soil samples were then frozen and brought to the lab for DNA extraction. Soil samples were sent to Waypoint Analytics to determine elemental composition of the soil and other factors such as organic matter content.

### ***DNA Extraction and Sequencing***
DNA extraction was carried out using the FastDNA Spin Kit for Soil. The DNA was then cleaned using Zymo Research DNA clean and concentrator because the initial DNA libraries sent were not easy to sequence and the cleaning kit was used to remove expected small, charged molecules from the DNA. The DNA was then packed into libraries and sent to Michigan State University and sequenced on a Novaseq 6000 S4 lane with paired-end 150 bp reads (University of Oregon GC3F facility).

### ***Taxonomic Identification with Metalign***

The quality scores of the received paired-end fastq files were examined to ensure clean sequencing. Once the files were evaluated, the paired-end sequence files were concatenated. The concatenated files were then uploaded to the William & Mary High Performance Cluster to run the identification software Metalign on each file using the mode “sensitive”. Metalign has been shown to classify taxa accurately and quickly (LaPierre et al., 2020).

### **Random Forest Modeling**

After the sequences were run through Metalign, the data were input into a Random Forest model for predicting whether any inflorescences were present on the _Asclepias_ sample based upon the input microbiome. A grid search was performed to identify the best out of this list of hyperparameters: taxonomic level, number of trees and max tree depth. The taxa in the dataset were also described by their percent presence across all samples and this became another hyperparameter labeled percent required. A grid search was conducted from 0% to 70% required identification across all samples in 10% increments.

The best performing model constituted 300 trees, with a max depth of 1, and all other hyperparameters were set to default. There was no increase in accuracy across the taxonomic levels, so phylum was used to shorten runtime. The optimum percent presence value was 0%. Therefore, the input data consisted of all 142 phyla identified, regardless of their rarity across all samples. Due to a class imbalance of 75% of the samples having at least one inflorescence, the parameter “class_weight” was set to “balanced_subsample”. This parameter will add weights to the Gini Impurity calculated in the tree based upon the composition of classes in the bootstrapped subset. No scaling or transformations were applied to the data.  The model was then put through Leave One Out cross-validation and train and test scores were calculated as averages across all k-folds.

## Results
### ***Description of Microbiome and Soil Data Results***
The dataset consisted of 156  plant samples with their number of inflorescences quantified and soil cores extracted. A large proportion of the plants had one or more inflorescences, 75%, or 116 samples, which created an imbalanced dataset with the remaining 25%, or 40 samples, having no inflorescences. Soil characteristics and plant measurements were found to differ significantly by site, with all one-way ANOVAs resulting in p<0.05. Collectively, 142 phyla, 164 classes, 372 orders, 836 families, 3,017 genera, and 22,645 species were identified across all soil samples using Metalign. The strain identifications were reported by Metalign; however, only the most accurately identified strain is reported. Therefore, the number of strains identified is artificially low as other strains of the same species identified at high accuracies would not be reported. The community composition when examining phyla across all samples was dominated by _Proteobacteria_ and _Actinobacteria_ (Figure 2).

![**Figure 2: Phylum community composition.** This figure contains two samples from each site and plots the relative abundance of each taxon. The communities were dominated in abundance by _Proteobacteria_ and _Actinobacteria_. The communities also displayed similar compositions throughout all samples except for three, which are not depicted here.](https://user-images.githubusercontent.com/67921793/167657065-51f29c0b-2d95-45f5-af29-d449d4c6d3e3.png)
**Figure 2: Phylum community composition.** This figure contains two samples from each site and plots the relative abundance of each taxon. The communities were dominated in abundance by _Proteobacteria_ and _Actinobacteria_. The communities also displayed similar compositions throughout all samples except for three, which are not depicted here.

The samples also displayed a similar distribution in species richness between zero inflorescences and one or more inflorescences plants across all taxonomic levels (Figure 3).

![**Figure 3: Species Richness for each taxonomic level comparing zero inflorescences and one or more inflorescences.** The distribution of species richness across samples is similar between zero and one or more inflorescences in all taxonomic levels.](https://user-images.githubusercontent.com/67921793/167658090-914b2171-84a7-4087-9714-f132921a35ac.png)
**Figure 3: Species Richness for each taxonomic level comparing zero inflorescences and one or more inflorescences.** The distribution of species richness across samples is similar between zero and one or more inflorescences in all taxonomic levels.

The level of phylum also displayed a high percentage of rarer taxa compared to the other taxonomic levels (Figure 4).

![**Figure 4: Percentage of representative of identified taxa across all samples for each taxonomic level.** The level of phylum displayed the largest variation in percent representation for each taxon across all samples and also contains the lowest average of around 22% average representation across all samples. One sample contained one identification of a viroid.](https://user-images.githubusercontent.com/67921793/167658412-ce0a8d01-a457-4b9f-8abb-489c25576523.png)
**Figure 4: Percentage of representative of identified taxa across all samples for each taxonomic level.** The level of phylum displayed the largest variation in percent representation for each taxon across all samples and also contains the lowest average of around 22% average representation across all samples. One sample contained one identification of a viroid.

The microbiome displayed high levels of multicollinearity, which could complicate analysis of important taxa (Figure 5) as correlation vs causation cannot be discerned by feature importance in the model.

![**Figure 5: Correlation matrix of all phyla.** Overall, some microbe groups were highly correlated with others while there also being numerous smaller correlations throughout. Interestingly, there were not many if any negative correlations seen.](https://user-images.githubusercontent.com/67921793/167658687-6956d977-3a75-46a7-9778-4833d81a8c35.png)
**Figure 5: Correlation matrix of all phyla.** Overall, some microbe groups were highly correlated with others while there also being numerous smaller correlations throughout. Interestingly, there were not many if any negative correlations seen.

### ***Description of Random Forest Modeling Results***
The resulting Random Forest Classifier model was put through leave-one-out cross validation on 156 observations, which contained relative abundance values of 142 phyla identified across the samples. To identify the optimal model parameters, a grid search was conducted across the following model and data parameters: number of trees, model depth, required percent presence across samples, and taxonomic level. The model parameters used in the final model consisted of 300 trees, a max depth of 1, zero taxa dropped from the data frame, and using the taxonomic level of phylum. The target variable, originally ranging from zero to ten, was changed to a binary variable where zero was all values equal to zero, and one if the values were above zero. This equates to predicting the presence of inflorescences. This binary classification resulted in an imbalanced dataset, with 75% of the data (n=116) samples having inflorescences and the rest (n=40) not having inflorescences. Through cross validation, the model had a training accuracy of 0.78% and testing accuracy of 0.64%. A confusion matrix was produced to describe the effectiveness of the model at predicting certain classes by their counts as well as by the percentage of that class overall (Figure 6).

![**Figure 6: Confusion matrices of Random Forest model predictions for zero and one or more inflorescences.** The matrix on the left details the count of class predictions by the Random Forest model. The model produced more false positive predictions than true negatives. The model was more accurate when predicting true positives overall, but still made many false negative predictions. These predictions are also reflected in the left matrix, which displayed the prediction counts as a percentage of the actual class.](https://user-images.githubusercontent.com/67921793/167677092-770df7ad-093f-429d-bb8d-66a58b9ec64a.png)
**Figure 6: Confusion matrices of Random Forest model predictions for zero and one or more inflorescences.** The matrix on the left details the count of class predictions by the Random Forest model. The model produced more false positive predictions than true negatives. The model was more accurate when predicting true positives overall, but still made many false negative predictions. These predictions are also reflected in the left matrix, which displayed the prediction counts as a percentage of the actual class.

Next, the important features of the model were determined by averaging the importance values across all 300 trees in the model for each fold in the leave-one-out cross-validation. Then, the importance values for each fold were also averaged. The result showed a clear top three phyla: _Euryarchaeota, Acidobacteria,_ and _Chlorobi,_ and 98 of 142 phyla had non-zero importance values (Figure 7)_._

![**Figure 7: Feature importance values of Random Forest model.** The top 13 most important features from the model ran show a clear top three most important phyla when predicting the presence of inflorescences.](https://user-images.githubusercontent.com/67921793/167677449-2e9b98ea-c504-4370-bafa-71d534451747.png)
**Figure 7: Feature importance values of Random Forest model.** The top 13 most important features from the model ran show a clear top three most important phyla when predicting the presence of inflorescences.

Due to variance in the ranking of important features when running the models using the same hyperparameters and different random states, the feature importances of 50 models through leave-one-out cross-fold validation were calculated to elucidate more clearly consistently important features. The importance features across all 300 trees were averaged for 50 models run through leave-one-out cross-fold validation and then those averages were averaged to determine the consistently important input features for the model. Of the 142 phyla in the dataset, 130 had an importance value greater than zero. The results for the top 13 informative taxa from the 50 models ran are reported in figure 8. The distribution of those importance values across all 142 phyla is reported in figure 9.

![**Figure 8: Important features in Random Forest model.** These are the top 13 features for predicting the presence of inflorescences averaged across 50 models. All values had a low level of importance, with the top three features slightly standing out from the rest.](https://user-images.githubusercontent.com/67921793/167677693-85440636-b88d-454e-a7f4-f80831580e47.png)
**Figure 8: Important features in Random Forest model.** These are the top 13 features for predicting the presence of inflorescences averaged across 50 models. All values had a low level of importance, with the top three features slightly standing out from the rest.

![**Figure 9: Distribution of important features.** The important features show the same top three and then a descent down in importance value.](https://user-images.githubusercontent.com/67921793/167677801-46037d4c-2f82-4154-a16f-74eb18041efa.png)
**Figure 9: Distribution of important features.** The important features show the same top three and then a descent down in importance value.

The abundance of the 13 most important phyla also exhibited strong correlations with one another (Figure 10).
![**Figure 10: Correlation matrix of important phyla in the random forest model.** Some phyla showed strong correlations with numerous other phyla, such as _Acidobacteria_.](https://user-images.githubusercontent.com/67921793/167677985-91fd917a-ff1f-4c65-888e-e24b425094ec.png)
**Figure 10: Correlation matrix of important phyla in the random forest model.** Some phyla showed strong correlations with numerous other phyla, such as _Acidobacteria_.

## **Discussion**

As discussed previously, soil microbiomes are reactive communities that respond to biotic and abiotic factors (Jiao et al., 2019; Luláková et al., 2019). This can present an issue when collecting soil samples for analysis, as the dynamic nature of the microbiome could result in different conditions being sampled across different sites. Sampling at numerous timepoints could account for such dynamism (Nannipieri et al., 2019).

A second consideration when sampling is the resilience of the different taxa to change. Bacterial abundances have been shown to vary in their resilience to environmental effects based upon their abundance, with taxa in large abundance resilient to change and rare taxa more sensitive (Jiao et al., 2019). The study also showed higher levels of resilience in communities with higher levels of diversity. If sampling at a site occurred right after a disturbance, then the rarer taxa could be missed as they were more susceptible to the change, resulting in lower community diversity in the sequenced microbial communities.

Along with the core communities around different locations of the plant, there are also predictable differences in microbiome diversity in the vertical distribution of microbe communities. Jiao et al. (2018) reported that as depth increases in the soil from 0 to 80 cm the bacterial and fungal diversity decreases, while the Archaeal diversity increases. Bacterial beta-diversity was linked to multi-nutrient cycling in the deep soil layers and the Archaeal beta-diversity was linked to the superficial soil layers. Therefore, the depth of the soil sample taken can influence the diversity of the sampled microbiome. Compared with the different depths of plant roots, standardized sampling practices need to be made to ensure the same area of the microbiome is captured. The bulk soil microbiome has also been shown to have a higher alpha diversity than the rhizosphere (Wu et al., 2018). Therefore, while collecting soil cores the researcher has to balance not gathering much root material while also minimizing the distance away from the roots to capture more of the rhizosphere than the bulk soil microbiome.

_Acidobacteria_ was one of the most important phyla in informing the model prediction. This phylum has been shown to increase in abundance as plant growth increases, but to also increase in abundance in the increasing abundance of PGPR (Kalam et al., 2017). _Acidobacteria_ abundance was correlated with nearly all other important taxa (_Euryarchaeota_: 0.15 and _Chlorobi_: 0.57). _Acidobacteria_ abundance  has also been shown to correlate with stages of plant development (Chaparro, Badri, & Vivanco, 2014). _Acidobacteria_ abundance increased from plant seed to vegetative developmental state, but then decreased once the plant started bolting and flowering.

_Euryarchaeota_ was identified as another important phylum in the model. _Euryarchaeota_ is a highly diverse phylum of Archaea that has taxa utilizing many different metabolic pathways (Bomberg & Timonen, 2007). _Euryarchaeota_ can be found in the rhizosphere and are not uncommonly the most abundant phylum. This taxon is thought to be correlated with the presence of mycorrhizal fungi and grows in older root tips over growing roots. When evaluating the correlations within the sampled microbiome, _Euryarchaeota_ had a correlation of 0.48 to the mycorrhizal fungi phylum _Mucoromycota._ Archaea have been shown through metagenomic analysis to influence the host plant in three ways: nutrient supply for the plant, promotion of plant growth via biosynthesis of auxin, and competition and syntrophic interactions with fungi and bacteria (Taffner et al., 2020).

The third most important phylum was _Chlorobi_. This phylum consists of green sulfur bacteria, most of which are phototrophic anaerobes and found in low-light environments (Bryant et al., 2012). Most members of this phylum can grow using only N2 as a source of Nitrogen and all require sulfide for their metabolism. This phylum has been identified as an endophyte (Kuffner et al., 2010); however, there is not much literature on its effects on plant growth. This phylum was possibly selected as important to model predictions because it is highly correlated with _Acidobacteria_ (0.57).

For future research, other plants characteristics may be predicted based upon the soil microbiome. The fitness of the plant may also be quantified based upon the plant’s characteristics and that value could be predicted as an overall view of how the microbiome affects the plant’s health. In the future, the researchers may switch to using regression models such as LASSO or ElasticNet. These models can remove features that do not contribute information to the prediction of the target, which is useful with datasets like in this study that have a plethora of features. Correlation terms would also be interesting to explore as certain microbes have been shown to interact with plants only in the presence of another specific taxa.

## **Conclusion**

As shown before, machine learning algorithms can detect patterns between microbiome community structures and host phenotypes, while also identifying important microbes within those communities as contributing the most predictive information. The important phyla for informing the model prediction have been shown to influence plant health and they have been shown to be affected by plant developmental stages (Chaparro, Badri, & Vivanco, 2014; Kalam et al., 2017; Taffner et al., 2020). In this study, the resulting soil microbiomes showed community compositions expected for the rhizosphere, with _Proteobacteria_ and _Actinobacteria_ dominating in abundance. Using a Random Forest model, the microbiome was able to contribute information in predicting the number of inflorescences. In the future, regression models might be used to yield better interpretations of taxa importance on model predictions. The important concept that microbes can predict host phenotypes underscores the importance of researching host microbiome relationships. In the case for agriculture, the soil microbiome can result in increased crop production by an environmentally sustainable method.
